cli/command/service/create.go

```diff
@@ -74,6 +74,8 @@ func runCreate(dockerCli *command.DockerCli, opts *serviceOptions) error {
 
 	ctx := context.Background()
 
+	resolveServiceImageDigest(dockerCli, &service)
```

Comment:
> Out of curiosity, docker build happens on a single worker node at a time, right? Does scheduling take into account which nodes have the image locally, or replicate the local images if the service needs to scale?

No, none of this is automatic. Typically a scenario using an image that's not pushed to a registry would be a single node swarm, or a few nodes where the image was manually copied in advance.